profile/src/wallet_kit_common/types/keys/is_private_key.rs

```diff
@@ -0,0 +1,14 @@
+use crate::prelude::*;
+
+pub trait IsPrivateKey<P: IsPublicKey<Self::Signature>>: Sized {
```

Comment:
currently only used by tests, so perhaps we should `cfg(test)` gate it...

Reply:
drawback of this impl is that call site I needed to take PrivateKey type AND PublicKey type as generic type:

```rust
  impl CAP26Vector {
        fn private_key<S, P>(&self) -> Result<S, CommonError>
        where
            P: IsPublicKey<S::Signature>
                + FromStr<Err = CommonError>
                + std::fmt::Debug
                + PartialEq,
            S: IsPrivateKey<P> + FromStr<Err = CommonError>,
        {
            let s: S = self.private_key_hex.parse()?;
            let p: P = self.public_key_hex.parse()?;
            assert_eq!(s.public_key(), p);
            Ok(s)
        }
    }
```

Reply:
I would prefer being able to just specify `S: IsPrivateKey`, without any more types, but it requires use to do `Self.Signature == PublicKey.Signature` which is not supported in stable rust...